rust/eventheader_macros/src/field_options.rs

```diff
@@ -0,0 +1,115 @@
+    Opt::new("hresult",                 I32_PATH,   E::Value32,             F::HexInt,      Scalar,     0),
+    Opt::new("hresult_slice",           I32_PATH,   E::Value32,             F::HexInt,      Slice,      0),
+    Opt::new("i16",                     I16_PATH,   E::Value16,             F::SignedInt,   Scalar,     0),
+    Opt::new("i16_hex",                 I16_PATH,   E::Value16,             F::HexInt,      Scalar,     0),
+    Opt::new("i16_hex_slice",           I16_PATH,   E::Value16,             F::HexInt,      Slice,      0),
+    Opt::new("i16_slice",               I16_PATH,   E::Value16,             F::SignedInt,   Slice,      0),
+    Opt::new("i32",                     I32_PATH,   E::Value32,             F::SignedInt,   Scalar,     0),
+    Opt::new("i32_hex",                 I32_PATH,   E::Value32,             F::HexInt,      Scalar,     0),
+    Opt::new("i32_hex_slice",           I32_PATH,   E::Value32,             F::HexInt,      Slice,      0),
+    Opt::new("i32_slice",               I32_PATH,   E::Value32,             F::SignedInt,   Slice,      0),
+    Opt::new("i64",                     I64_PATH,   E::Value64,             F::SignedInt,   Scalar,     0),
+    Opt::new("i64_hex",                 I64_PATH,   E::Value64,             F::HexInt,      Scalar,     0),
+    Opt::new("i64_hex_slice",           I64_PATH,   E::Value64,             F::HexInt,      Slice,      0),
+    Opt::new("i64_slice",               I64_PATH,   E::Value64,             F::SignedInt,   Slice,      0),
+    Opt::new("i8",                      I8_PATH,    E::Value8,              F::SignedInt,   Scalar,     0),
+    Opt::new("i8_hex",                  I8_PATH,    E::Value8,              F::HexInt,      Scalar,     0),
+    Opt::new("i8_hex_slice",            I8_PATH,    E::Value8,              F::HexInt,      Slice,      0),
+    Opt::new("i8_slice",                I8_PATH,    E::Value8,              F::SignedInt,   Slice,      0),
+    Opt::new("ipv4",                    U8_PATH,    E::Value32,             F::IPv4,        Scalar,     4),
+    Opt::new("ipv4_slice",              U8_PATH,    E::Value32,             F::IPv4,        Slice,      4),
+    Opt::new("ipv6",                    U8_PATH,    E::Value128,            F::IPv6,        Str,        16),
+    Opt::new("ipv6c",                   U8_PATH,    E::Value128,            F::IPv6,        Str,        16),
+    Opt::new("isize",                   ISIZE_PATH, E::ValueSize,           F::SignedInt,   Scalar,     0),
+    Opt::new("isize_hex",               ISIZE_PATH, E::ValueSize,           F::HexInt,      Scalar,     0),
+    Opt::new("isize_hex_slice",         ISIZE_PATH, E::ValueSize,           F::HexInt,      Slice,      0),
+    Opt::new("isize_slice",             ISIZE_PATH, E::ValueSize,           F::SignedInt,   Slice,      0),
+    Opt::new("pid",                     U32_PATH,   E::Value32,             F::Pid,         Scalar,     0),
```

Comment:
Probably would be worthwhile to have linux_time32, linux_time64, and linux_errno. (Or should it be linux_error?)